packages/ra-core/src/auth/useLogout.ts

```diff
@@ -27,20 +26,8 @@ import { LocationDescriptorObject } from 'history';
 const useLogout = (): Logout => {
     const authProvider = useAuthProvider();
     const dispatch = useDispatch();
-
-    /**
-     * We need the current location to pass in the router state
-     * so that the login hook knows where to redirect to as next route after login.
-     *
-     * But if we used useLocation to get it, the logout function
-     * would be rebuilt each time the user changes location. Consequently, that
-     * would force a rerender of all components using this hook upon navigation
-     * (CoreAdminRouter for example).
-     *
-     * To avoid that, we read the location directly from history which is mutable.
-     * See: https://reacttraining.com/react-router/web/api/history/history-is-mutable
-     */
-    const history = useHistory();
+    const navigate = useNavigate();
+    const location = useLocation();
```

Comment:
Major perf problem, as locaiton is in the useCallback signature, the logout callback will change each time the user navigates, and the Logout button will be rerendered.

Reply:
Fixed using a location ref